File: 12_28_SpringMVC_04/src/main/java/com/zelin/service/StudentService.java
package com.zelin.service;

import com.zelin.pojo.PageResult;
import com.zelin.pojo.Student;

public interface StudentService {
    //分页条件查询
    PageResult search(Integer page, Integer pageSize, Student student);
    //添加学生
    void addStu(Student student);
    //删除学生
    void deleteStu(String sid);

    //修改学生
    void updateStu(Student student);
}

File: 12_30_SpringMVC_05/src/main/webapp/js/service/studentService.js
app.service("studentService", function ($http) {
    this.search = (page, pageSize, entity) => {
        return $http.post("student/search.do?page=" + page + "&pageSize=" + pageSize, entity);
    }


    this.save=(url,stuEntity) =>{
        return $http.post(url,stuEntity);
    }

    this.deleteStu = (sid) => {
        return $http.get("student/delete.do?sid="+sid);
    }

    this.deleteAll = (sids) => {
        return $http.get("student/deleteAll.do?sids="+sids);
    }
});